### ISSUE TITLE
Adding advanced object ownership system

### ISSUE THREAD

**Issue by IvanMMM** [micovery/ArmA3_Wasteland.Altis#39]
Hey. It would be nice if we could add some kind of permission system to gain access to object, just like with cameras or spawn beacons, but more general. 
Allow options must be: Only for me, For group, For team, for specific players.
I think it must include global permissions for all objects/vehicles you own and permission settings for each object you own (list or action on object) and ability to change owner.
I will help if I can :)

**Comment by micovery** [comment]
Sounds interesting ... but what does it mean to "own" an object. What actions can the object "owner" (or owners) do that other's cant?  Also, how would the ownership of the "objects" be enforced?

**Comment by IvanMMM** [comment]
Owning object allows player to preform any actions like unlocking, moving, towing, lifting + setup access properties.
Player with permission can preform any allowed action on the object.
Player without permission cant do anything with this object. Only destroy.
I mean some players really don't want to anyone can move their baseparts. That's the main idea. Do useful permission system.

And about ownership. In vanilla system person who locked object is it's owner.

**Comment by micovery** [comment]
I see, makes sense. It should be configurable as to what kinds of objects  use the permission system. 

I can imagine that some admins would not want to apply it to everything. For example, static weapons  ... it would be awkward if a "static weapon" was locked, and you could not get into it.

**Comment by IvanMMM** [comment]
Yeah, sure. It should can be easily configured or disabled at all.

**Comment by Cael817** [comment]
This was mentioned on the A3W forums maybe by one of you, don't remember but something like Epochs "Frequency Jammer" (if we are talking about bases and such things), this is also something that players ask for alot. and we could have pretty short times for saving baseparts unless players place something like this in their base. I think it also act as a stimulus to the economy and cleaning the server of unused baseparts. Having to pay an upkeep while your baseparts is protected seem to be an interest of many. If being able to lock doors/boxes and whatnot toghether with this basebuilding would probably be much more frequent. But if they get this they want the whole building system from Epoch ;-).

Im guessing this might involve alot of coding but like LouDnl do in his R3F and with the "donator" part maybe it wont have to be that complex.

https://github.com/LouDnl/ArmA3_Wasteland.Altis/tree/extDB_beta_R3F3.1
http://epochmod.gamepedia.com/A2Epoch:Frequency_Jammer
